libraries/AP_NavEKF3/AP_NavEKF3.cpp

```diff
@@ -1083,6 +1083,9 @@ void NavEKF3::resetCoreErrors(void)
 // set position, velocity and yaw sources to either 0=primary, 1=secondary, 2=tertiary
 void NavEKF3::setPosVelYawSourceSet(uint8_t source_set_idx)
 {
+    if (source_set_idx < 3) {
```

Comment:
AP_NAKEKF_SOURCE_SET_MAX could be used instead of "3" although I can imagine you might not want to use it because of the limitation on the AP_DAL event definitions.  If you used it you might also need to add an assert.

Reply:
fixed